lib/ExpensiMark.js

```diff
@@ -60,15 +60,20 @@ export default class ExpensiMark {
                 name: 'email',
                 process: (textToProcess, replacement) => {
                     const regex = new RegExp(
-                        `(?!\\[\\s*\\])\\[([^[\\]]*)]\\(${CONST.REG_EXP.MARKDOWN_EMAIL}\\)`, 'gim'
+                        `(?!\\[\\s*\\])\\[([^[\\]]*)]\\((mailto:)?${CONST.REG_EXP.MARKDOWN_EMAIL}\\)`, 'gim'
                     );
                     return this.modifyTextForEmailLinks(regex, textToProcess, replacement);
                 },
                 replacement: (match, g1, g2) => {
                     if (g1.match(CONST.REG_EXP.EMOJIS) || !g1.trim()) {
                         return match;
                     }
-                    return `<a href="mailto:${g2}">${g1.trim()}</a>`;
+                    let label = g1.trim();
+                    const href = `mailto:${g2}`;
+                    if (label === href) {
+                        label = g2;
+                    }
+                    return `<a href="${href}">${label}</a>`;
```

Comment:
I don't like the fact that we mutate `label` here but honestly I cannot think of anything better that wouldn't require duplicating `<a href="...">` string.